vermouth/rcsu/contact_map.py

```diff
@@ -12,10 +12,610 @@
+    G = make_residue_graph(system.molecules[0])
+
+    resids = []
+    chains = []
+    resnames = []
+    positions_all = []
+    cogs = []
+    ca_pos = []
+    vdw_list = []
+    atypes = []
+    res_serial = []
+    nodes = []
+    for node in G.nodes:
+        # we only need these for writing at the end
+        resnames.append(G.nodes[node]['resname'])
+        resids.append(G.nodes[node]['resid'])
+        chains.append(G.nodes[node]['chain'])
+        nodes.append(G.nodes[node]['_res_serial'])
+
+        res_pos = []
+        for subnode in sorted(G.nodes[node]['graph'].nodes):
+            if 'position' in G.nodes[node]['graph'].nodes[subnode]:
+                res_serial.append(G.nodes[node]['graph'].nodes[subnode]['_res_serial'])
+
+                res_pos.append(G.nodes[node]['graph'].nodes[subnode]['position'] * 10)
+                positions_all.append(G.nodes[node]['graph'].nodes[subnode]['position'] * 10)
+
+                vdw_list.append(get_vdw_radius(G.nodes[node]['graph'].nodes[subnode]['resname'],
+                                               G.nodes[node]['graph'].nodes[subnode]['atomname']))
+                atypes.append(get_atype(G.nodes[node]['graph'].nodes[subnode]['resname'],
+                                        G.nodes[node]['graph'].nodes[subnode]['atomname']))
+
+            if G.nodes[node]['graph'].nodes[subnode]['atomname'] == 'CA':
+                ca_pos.append(G.nodes[node]['graph'].nodes[subnode]['position'])
+
+        cogs.append(np.stack(np.tile(np.stack(res_pos).mean(axis=0), (len(res_pos), 1))))
+
+    cogs = np.vstack(cogs)
+    vdw_list = np.array(vdw_list)
+    atypes = np.array(atypes)
+    coords = np.stack(positions_all)
+    res_serial = np.array(res_serial)
+
+    resids = np.array(resids)
+    chains = np.array(chains)
+    resnames = np.array(resnames)
+    nodes = np.array(nodes)
+
+    # 2) find the number of residues that we have
+    nresidues = len(G)
+
+    return cogs, vdw_list, atypes, coords, res_serial, resids, chains, resnames, nodes, ca_pos, nresidues, G
+
+
+def calculate_contact_map(cogs, vdw_list, atypes, coords, res_serial,
+                          resids, chains, resnames, nodes, ca_pos, nresidues,
+                          G):
+
+    # some initial definitions of variables that we need
+    fib = 14
+    fiba, fibb = 0, 1
+    for _ in range(fib):
+        fiba, fibb = fibb, fiba + fibb
+
+    alpha = 1.24  # Enlargement factor for attraction effects
+    water_radius = 2.80  # Radius of a water molecule in A
+
+    # 4) set up final bits of information
+    # sums of vdw pairs for each atom we have
+    vdw_sum_array = vdw_list[:, np.newaxis] + vdw_list[np.newaxis, :]
+
+    # distances between all the atoms that we have
+    atomic_distances = cdist(coords, coords)
+
+    # array with 1 on the diagonal, so we can exclude the self atoms
+    diagonal_ones = np.diagflat(np.ones(atomic_distances.shape[0], dtype=int))
```

Comment:
```suggestion
    diagonal_ones = np.diagflat(np.full(atomic_distances.shape[0], np.inf))
```
Maybe? Saves a check later on. But I think this goes away with the kdtree anyway